specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json

```diff
@@ -518,6 +567,23 @@
       ],
       "description": "Tenant Ids information."
     },
+    "CheckResourceNameResult": {
+      "properties": {
+        "name": {
+          "description": "Name of Resource",
+          "type": "string"
+        },
+        "type": {
+          "description": "Type of Resource",
+          "type": "string"
+        },
+        "status": {
+          "description": "Is the resource name allowed or not",
+          "type": "string"
+        }
+      },
+      "description": "Validity of Resource name for the given type"
```

Comment:
Note - might be worth explaining what this *doesn't* do. For example, `Microsoft.Storage` do their own validation of names and have stricter requirements as they must be globally unique. I wouldn't want customers to think they can use this API to perform the same check.